File: ruby/lib/zser.rb
# frozen_string_literal: true

require "zser/version"
require "zser/exceptions"

require "zser/decoder"
require "zser/object"
require "zser/parser"
require "zser/varint"

# zcred serialization format
module Zser
  # Parse the given self-describing zser message
  #
  # @param message [String] binary encoded zser message
  #
  # @return [Zser::Object] Hash-like object representing message
  def self.parse(message)
    parser = Zser::Parser.new(Zser::Decoder.new)
    parser.parse(message)
    parser.finish
  end
end

File: js/src/zser.ts
import { Decoder } from "./decoder";
import { Parser } from "./parser";

export default class Zser {
  // Parse a Uint8Array containing an encoded zser message to an object representation
  public static parse(message: Uint8Array): object {
    const parser = new Parser(new Decoder());
    parser.parse(message);
    return parser.finish();
  }
}
